[MIG] event_sale_registration_multi_qty: Migration to 14.0

chienandalu opened the pull request:
In this migration:

- Adapt to new methods and behaviors of the sale order in relation to the registrations.
- Fix the registrations count to sum multi qty registrations.

- [x] Pending on #265 

cc @Tecnativa TT34349

please review @ernestotejeda @pedrobaeza

chienandalu committed:
[10.0][NEW] event_sale_registration_multi_qty: New module

chienandalu committed:
[12.0] event_sale_registration_multi_qty: Migration to 12.0

CarlosRoca13 committed:
[IMP] event_sale_registration_multi_qty: black, isort

CarlosRoca13 committed:
[MIG] event_sale_registration_multi_qty: Migration to v13.0

victoralmau committed:
[IMP] event_sale_registration_multi_qty: Compatibility with event_sale_reservation addon.
TT31321

chienandalu committed:
[IMP] event_sale_registration_multi_qty: black, isort, prettier

chienandalu committed:
[MIG] event_sale_registration_multi_qty: Migration to 14.0

TT34349

pedrobaeza commented:
/ocabot migration event_sale_registration_multi_qty

pedrobaeza commented on event_sale_registration_multi_qty/tests/test_event_sale_registration_multi_qty.py:
```diff
@@ -36,72 +35,44 @@ def setUpClass(cls):
+    def _create_sale(self):
+        sale_form = Form(self.env["sale.order"])
+        sale_form.partner_id = self.partner
+        return sale_form.save()
```
For being a bit faster, return the form and don't save until the end.

pedrobaeza commented on event_sale_registration_multi_qty/wizards/event_edit_registration.xml:
```diff
@@ -7,7 +7,7 @@
         <field name="arch" type="xml">
             <field name="name" position="before">
-                <field name="qty" />
+                <field name="qty" readonly="1" />
```
force_save=1?

chienandalu committed:
[FIX] event_registration_multi_qty: sane default

If we force the multi qty mode by default there's a high chance to come
across unexpected CI errors.

pedrobaeza commented:
/ocabot merge nobump
